src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/RerankingMetrics.java

```diff
@@ -0,0 +1,56 @@
+package io.stargate.sgv2.jsonapi.service.operation.reranking;
+
+import static io.stargate.sgv2.jsonapi.util.ClassUtils.classSimpleName;
+
+import io.micrometer.core.instrument.*;
+import io.stargate.sgv2.jsonapi.api.request.RequestContext;
+import io.stargate.sgv2.jsonapi.service.reranking.operation.RerankingProvider;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class RerankingMetrics {
+  final String RERANKING_NUMBER_OF_PASSAGE = "reranking.passage.count";
+  final String RERANKING_CALL_DURATION_METRICS = "reranking.call.duration";
+  final String RERANKING_PROVIDER_METRICS_TAG = "reranking.provider";
+  final String RERANKING_PROVIDER_MODEL_METRICS_TAG = "reranking.model";
+  final String TENANT_TAG = "tenant";
+  final String UNKNOWN_VALUE = "unknown";
+  private final MeterRegistry meterRegistry;
+  private final RerankingProvider rerankingProvider;
+  private final RequestContext requestContext;
+
+  public RerankingMetrics(
```

Comment:
this has a public ctor, but no public functions . Check the visibility for all functions is correct.